src/Infer/Scope/Scope.php

```diff
@@ -52,6 +53,13 @@ public function getType(Node $node): Type
             return (new ConstFetchTypeGetter)($node);
         }
 
+        if ($node instanceof Node\Expr\Ternary) {
+            return new Union([
+                $this->getType($node->if),
+                $this->getType($node->else)
+            ]);
```

Comment:
There is `Union::wrap` method that will generate `Union` type in a smart way (it'll remove duplicated types etc):

```suggestion
            return Union::wrap([
                $this->getType($node->if),
                $this->getType($node->else)
            ]);
```